src/topics/topicSelectors.js

```diff
@@ -66,8 +66,8 @@ export const getTopicsForStream: Selector<?(TopicExtended[]), number> = createSe
   },
 );
 
-export const getLastMessageTopic = (narrow: Narrow): Selector<string> =>
-  createSelector(
-    state => getShownMessagesForNarrow(state, narrow),
-    messages => (messages.length === 0 ? '' : messages[messages.length - 1].subject),
-  );
+export const getLastMessageTopic: Selector<string, Narrow> = createSelector(
+  (state, narrow) => narrow,
+  (state, narrow) => getShownMessagesForNarrow(state, narrow),
+  (narrow, messages) => (messages.length === 0 ? '' : messages[messages.length - 1].subject),
+);
```

Comment:
This is an example where the caching isn't doing anything useful, even if we fix it so the caching happens. That's because the body isn't doing anything expensive: it's doing O(1) work. So this can turn into a plain function with no `createSelector`.